File: CodingTestPlatform/BOJ/2_Silver/3/11659/11659.py
# 구간 합 구하기 4
# https://www.acmicpc.net/problem/11659

import sys
input = sys.stdin.readline

t = list(map(int, input().split()))

arr = list(map(int, input().split()))
acc = [arr[0]]

for i in range(t[0]):
    if i > 0:
        acc.append(arr[i] + acc[i - 1])

for i in range(t[1]):
    start, end = list(map(lambda num: num - 1, map(int, input().split())))
    print(acc[end] - acc[start - 1]) if start !=0 else print(acc[end])

File: CodingTestPlatform/BOJ/2_Silver/3/11659/11659.js
/**
 * 구간 합 구하기 4
 * Silver3
 * https://www.acmicpc.net/problem/11659
 */

const [NM, number, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const numArr = number.split(" ").map((c) => parseInt(c));
let accSum = [0];

for (let i = 0; i < numArr.length; i++) {
    accSum.push(accSum[i] + numArr[i]);
}

const result = input.map((v) => {
    const [start, end] = v.split(" ");
    return accSum[end] - accSum[start-1];
});

console.log(result.join("\n"));